src/main/javassist/util/proxy/ProxyFactory.java

```diff
@@ -1203,6 +1209,24 @@ private static boolean isOverloaded(int index, Method[] methods) {
 
         return false;
     }
+    
+    private static boolean isBridgeAndOverloaded(Method meth, List methods) {
+        if (isBridge(meth)) {
```

Comment:
You see why I was raising this as ambiguous, we already know it is not inBridge from passing the previous or condition from the original if statement. But in this context yes you need to check it again to be complete adding extra overhead.

Reply:
If want to remove this extra overhead, maybe below method is needed:
isNotBridgeOrBridgeAndOverloadedAndPublic()
The method name is too long and not easy to understand.